src/general/MainMenu.tscn

```diff
@@ -11,6 +11,12 @@
 [ext_resource path="res://assets/textures/gui/BG_Menu03.png" type="Texture" id=9]
 [ext_resource path="res://assets/textures/gui/logo.png" type="Texture" id=10]
 
+[sub_resource type="InputEventKey" id=2]
+scancode = 16777217
+
+[sub_resource type="ShortCut" id=3]
+shortcut = SubResource( 2 )
```

Comment:
Our custom input system should be used, otherwise the keys can't be rebound properly through the options menu.

Reply:
Is that really needed? I don't think anyone will use any key else though.

Anyway I will try.

Reply:
What about controller support? That would need specific handling here if it doesn't go through the input system.

Reply:
Unable to understand

I need to add a new keybinding but failed

Reply:
You don't need a new keybinding as there is already the `ui_cancel` action in Godot.
That can be used through the input system with this attribute: `[RunOnKeyDown("ui_cancel")]`

Reply:
Still Useless?

Reply:
What is useless?
You can look at how the pause menu closes itself, and copy that for going back to the main menu.
Also I just realized that you can't put the key usage in the OptionsMenu because that is also used by the pause menu so that means both want to react to the ESC press, which is not good design.

Reply:
Instead I'd put the logic in the main menu class to switch back to the top most level menu on ESC being pressed, this way it'll work for also being in the tools sub menu.